src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts

```diff
@@ -60,21 +59,25 @@ export class TerminalProfileService implements ITerminalProfileService {
 		// that contributes a profile
 		this._extensionService.onDidChangeExtensions(() => this.refreshAvailableProfiles());
 
-		this._configurationService.onDidChangeConfiguration(async e => {
-			const platformKey = await this.getPlatformKey();
-			if (e.affectsConfiguration(TerminalSettingPrefix.DefaultProfile + platformKey) ||
-				e.affectsConfiguration(TerminalSettingPrefix.Profiles + platformKey) ||
-				e.affectsConfiguration(TerminalSettingId.UseWslProfiles)) {
-				this.refreshAvailableProfiles();
```

Comment:
`affectsConfiguration` wasn't working as expected. even when `e.affectedKeys` were totally unrelated to the terminal/profiles, it was entering this statement and refreshing the available profiles

Reply:
investigating why that is